lib/reek/tree_walker.rb

```diff
@@ -211,6 +218,14 @@ def visibility_modifier?(call_node)
       VISIBILITY_MODIFIERS.include?(call_node.method_name)
     end
 
+    # FIXME: Move to SendNode?
+    def attribute_writer?(call_node)
+      method_name = call_node.method_name
+      ATTR_DEFN_METHODS.include?(method_name) ||
+        method_name == :attr && call_node.args.last.type == :true
```

Comment:
Hmmm, can we somehow put 

```Ruby
method_name == :attr && call_node.args.last.type == :true
```

into a separate, appropriately named method and in this have method explaining this? I assume that most people are not familiar with 

```Ruby
attr :foo, true
```

How about

```Ruby
def attr_call_special_form(method_name, call_node)
  # Handles the case where we create an attribute writer via:
  # attr :foo, true
  method_name == :attr && call_node.args.last.type == :true
end
```

?